Python/symtable.c

```diff
@@ -2298,16 +2300,35 @@ symtable_visit_expr(struct symtable *st, expr_ty e)
 }
 
 static int
-symtable_visit_type_param_bound_or_default(struct symtable *st, expr_ty e, identifier name, void *key)
+symtable_visit_type_param_bound_or_default(
+    struct symtable *st, expr_ty e, identifier name, void *key,
+    const char *tp_kind_name, int is_default)
 {
     if (e) {
         int is_in_class = st->st_cur->ste_can_see_class_scope;
-        if (!symtable_enter_block(st, name, TypeVarBoundBlock, key, LOCATION(e)))
+        if (!symtable_enter_block(st, name, TypeVariableBlock, key, LOCATION(e)))
             return 0;
+
         st->st_cur->ste_can_see_class_scope = is_in_class;
         if (is_in_class && !symtable_add_def(st, &_Py_ID(__classdict__), USE, LOCATION(e))) {
             VISIT_QUIT(st, 0);
         }
+
+        PyObject *scope_info;
+        scope_info = PyUnicode_FromFormat("%s %s", tp_kind_name,
```

Comment:
Would be simpler to pass the whole `ste_scope_info` string into this function so you don't have to create it dynamically.

The current code appears to be leaking the value of `scope_info`. (Tip: run `./python -m test -R 3:3 test_type_params`, or some other test, to check for refleaks.)

Reply:
Oh yes thank you. I actually struggled a bit to understand how to create a string and whether there is a mechanism for deallocating them correctly (maybe you could point some section I could read about that?).

By the way, on the main branch I have:

```text
Using random seed: 831335327
0:00:00 load avg: 2.20 Run 1 test sequentially in a single process
0:00:00 load avg: 2.20 [1/1] test_type_params
beginning 6 repetitions. Showing number of leaks (. for 0 or less, X for 10 or more)
123:456
XX. ...
```

Is it fine that there are leaks? (the test does not fail but I would like to know why and whether I should bother).

Reply:
Leaks on the first two ("warmup") runs are usually expected; various caches are being filled up. We care that there are no leaks on the last few runs.

Reply:
As for deallocating them, you're creating a PyObject, so just `Py_DECREF` drops the reference and deallocates the string you've created.

Reply:
Ah yes. Actually, I know what I did wrong. I think I read "The caller is **NOT** responsible for decref'ing the returned objects." and I was like "Oh great!"...